### PR TITLE
Fall back to CPU when parsing two-digit years

### LINKED ISSUE #2118: [BUG] to_date with two digit year uses different century between CPU and GPU
**Describe the bug**
I enabled `spark.rapids.sql.incompatibleDateFormats.enabled` and attempted to parse `31/12/99` using the format `dd/MM/yy`. 

Spark on CPU returns `2099-12-31` but Spark on GPU returns `1999-12-31`.

**Steps/Code to reproduce bug**

Update `ParseDateTimeSuite.scala`:

- Add value `"31/12/99"` to `timestampValues` variable 

- Add new test:

```scala
testSparkResultsAreEqual("to_date dd/MM/yy",
  datesAsStrings,
  conf = new SparkConf().set(SQLConf.LEGACY_TIME_PARSER_POLICY.key, "CORRECTED")) {
  df => df.withColumn("c1", to_date(col("c0"), "dd/MM/yy"))
}
```

**Expected behavior**
Test should pass or we should fall back to CPU for `yy`.

**Additional context**
Spark 3.1.1

### PR DESCRIPTION
**Pull request by andygrove**
Signed-off-by: Andy Grove <andygrove@nvidia.com>

Closes https://github.com/NVIDIA/spark-rapids/issues/2127

### CODE REVIEW

**Review by jlowe — DISMISSED** [review]

**Review by jlowe — DISMISSED** [review]

**Review by revans2 — DISMISSED** [review]

**Comment by sameerz** [comment]
build

**Review by gerashegalov — DISMISSED** [review]

**Comment by andygrove** [comment]
Test failed in CI against Spark 3.0.0 - I am investigating

**Comment by andygrove** [comment]
build

**Comment by andygrove** [comment]
@jlowe @gerashegalov @revans2 So I had to make further changes on this PR because we do support "yy" correctly in some contexts (when not parsing dates from strings). Tests pass now and this is ready for another review.

File: docs/compatibility.md [hunk 1]
```diff
@@ -256,7 +256,6 @@ The formats which are supported on GPU and 100% compatible with Spark are :
 
 Examples of supported formats that may produce different results are:
 
-- Use of a two digit year, i.e.: `yy`, may use a different century on GPU compared to CPU
```
Comment (revans2) [inline docs/compatibility.md:259]: Should this be updated to explain that yy is not supported when pasting a string, but in other cases it is?

Comment (andygrove) [reply docs/compatibility.md:259]: This line was removed from the documentation for parsing strings, and I have now added a new section for" formatting dates as strings" and have pointed out that "yy" is supported for formatting but not for parsing.

**Review by revans2 — APPROVED** [review]

**Comment by revans2** [comment]
build

**Review by jlowe — APPROVED** [review]

**Review by gerashegalov — APPROVED** [review]

**Pull request merged**